okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -131,6 +133,9 @@ public void onResponse(Call _call, Response response) {
             return future.get();
         } catch (InterruptedException e) {
             getDelegate().cancel();
+            if (!future.setException(e)) {
+                closeResponseBody(future);
+            }
             Thread.currentThread().interrupt();
             throw new InterruptedIOException("Call was interrupted during execution");
         } catch (ExecutionException e) {
```

Comment:
I don't believe you ever get a leak here, because this can only be set when the response fails via the callback. We *do* set the exception of the future inside the interruption handling, but we're already inside a catch at the point and we've stopped blocking on the `get`.